src/hotspot/cpu/aarch64/aarch64_vector_ad.m4

```diff
@@ -1779,11 +1776,20 @@ dnl
 dnl REDUCE_ADD_FP_SVE($1,   $2  )
 dnl REDUCE_ADD_FP_SVE(type, size)
 define(`REDUCE_ADD_FP_SVE', `
+// This rule calculates the reduction result in strict order. Two cases will
+// reach here:
+// 1. Non strictly-ordered AddReductionV$1 when vector size > 128-bit. Such
+//    nodes can only be generated by Vector API.
+// 2. Strictly-ordered AddReductionV$1, which can only be generated by
+//    auto-vectorization on SVE machine.
 instruct reduce_add$1_sve(vReg$1 dst_src1, vReg src2) %{
-  predicate(UseSVE > 0);
+  predicate((!VM_Version::use_neon_for_vector(Matcher::vector_length_in_bytes(n->in(2))) &&
+             !n->as_Reduction()->requires_strict_order()) ||
+            n->as_Reduction()->requires_strict_order());
```

Comment:
Same as above. Can it be simplified to `!VM_Version::use_neon_for_vector(Matcher::vector_length_in_bytes(n->in(2))) || n->as_Reduction()->requires_strict_order()`?